src/cmds/testing/block_dev/block_dev_test.c

```diff
@@ -499,13 +532,26 @@ int main(int argc, char **argv) {
 				n_blocks = strtoll(optarg, NULL, 0);
 				n_blocks_flag = 1;
 				break;
+		        case 'm':
+				m_blocks = strtoll(optarg, NULL, 0);
+				m_blocks_flag = 1;
+				break;
 			case 'h':
 			default:
 				print_help();
 				return 0;
 		}
 	}
 
+	if (m_blocks_flag) {
+		if (m_blocks < 0) {
+			printf("Multiple blocks count argument value should be greater than zero\n");
+			return -EINVAL;
+		} else if (!m_blocks) {
+			m_blocks = DEFAULT_MULTIPLE_BLOCKS_NUMBER;
+		}
```

Comment:
Maybe interpret `-m 0` like an error too? So we do not need DEFAULT_MULTIPLE_BLOCKS_NUMBER then and can just use `m_blocks = 1` if `-m` wasn't specified.

Reply:
Or is it possible to provide the `-m` flag without an argument at all?

Reply:
If we provide -m flag without an argument, and if we try to read the argument we get it as zero. So, i added that case. As Deryugin said 

> have some default value so we don't require user to specify it every time.

The idea is this: In General, the command runs with single block read/write per iteration. If the user calls the command with '-m' flag and without an argument (or) zero argument, it runs with the default block read/writes per iteration. If the user specifies an argument, then the command runs with that many read/writes per iteration.

Reply:
I want to point here that calling `-m` without an argument shouldn't be allowed at all. When you say that user can run just `-m` you should guarantee it would be handled correctly anyway. Let's see a simple program:

```
#include <stdio.h>
#include <unistd.h>
#include <stdlib.h>

int main(int argc, char *argv[]) {
    int opt;
    int targ, tfnd;

    while ((opt = getopt(argc, argv, "t:")) != -1) {
        switch (opt) {
        case 't':
            targ = atoi(optarg);
            tfnd = 1;
            break;
        default: /* '?' */
            fprintf(stderr, "Usage: %s [-t arg] name\n",
                argv[0]);
            exit(EXIT_FAILURE);
        }
    }
    printf("tfnd=%d; targ=%d; name=%s\n",
        tfnd, targ, argv[optind]);

    exit(EXIT_SUCCESS);
}
```

```
$ gcc - o test test.c

$ ./test -t 7 hda
tfnd=1; targ=7; name=hda

$ ./test -t
./test: option requires an argument -- 't'
Usage: ./test [-t arg] name

$ ./test -t 1231.hda
tfnd=1; targ=1231; name=(null)
```

So you can see that value of `targ` actually depends on what use passed. And if you allow `-t` without arguments, then in the last case `./test -t 1231.hda` the input is pretty correct but you see what happened..

Reply:
One more example when input is correct, but the name becomes null.

```
$ ./test -t hda
tfnd=1; targ=0; name=(null)
```

Reply:
I was wondering why block_dev_test works in your branch when I run it as following:

```
root@embox:/#block_dev_test -m sda
```

But it's actually because of this:

```
bdev = block_dev_find(argv[argc - 1]);
```

so in this case `-m` parse `sda` and then we select sda also like a block device name. It doesn't seem to be an expected behavior.

Reply:
This is a very good find @alexkalmuk. I learnt something new, thanks. I will update with a new commit.

Reply:
> I was wondering why block_dev_test works in your branch when I run it as following:
> 
> ```
> root@embox:/#block_dev_test -m sda
> ```
> 
> But it's actually because of this:
> 
> ```
> bdev = block_dev_find(argv[argc - 1]);
> ```
> 
> so in this case `-m` parses `sda` and then we select sda also like a block device name. It doesn't seem to be an expected behavior.

Yes, i will solve this in another PR (i am keeping this PR to solve issue #1709). Is that ok?

Reply:
> I was wondering why block_dev_test works in your branch when I run it as following:
> 
> ```
> root@embox:/#block_dev_test -m sda
> ```
> 
> But it's actually because of this:
> 
> ```
> bdev = block_dev_find(argv[argc - 1]);
> ```
> 
> so in this case `-m` parses `sda` and then we select sda also like a block device name. It doesn't seem to be an expected behavior.

Yes, you are right. This is not an expected behavior, we can solve this by using second argument of `strtol` to check whether the input was valid. But, i would like to take up on that in another PR, as this PR is specifically for solving #1709. Is that ok?